src/Libraries/Microsoft.Extensions.DataIngestion/Processors/ClassificationEnricher.cs

```diff
@@ -0,0 +1,95 @@
+/// an optional fallback class for cases where no suitable classification can be determined.</remarks>
+public sealed class ClassificationEnricher : IngestionChunkProcessor<string>
+{
+    private readonly IChatClient _chatClient;
+    private readonly ChatOptions? _chatOptions;
+    private readonly TextContent _request;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ClassificationEnricher"/> class.
+    /// </summary>
+    /// <param name="chatClient">The chat client used for classification.</param>
+    /// <param name="predefinedClasses">The set of predefined classification classes.</param>
+    /// <param name="chatOptions">Options for the chat client.</param>
+    /// <param name="fallbackClass">The fallback class to use when no suitable classification is found. When not provided, it defaults to "Unknown".</param>
+    public ClassificationEnricher(IChatClient chatClient, ReadOnlySpan<string> predefinedClasses,
+        ChatOptions? chatOptions = null, string? fallbackClass = null)
+    {
+        if (predefinedClasses.Length == 0)
+        {
+            Throw.ArgumentException(nameof(predefinedClasses), "Predefined classes must be provided.");
+        }
+
+        _chatClient = Throw.IfNull(chatClient);
+        _chatOptions = chatOptions;
+        _request = CreateLlmRequest(predefinedClasses, string.IsNullOrEmpty(fallbackClass) ? "Unknown" : fallbackClass!);
+    }
+
+    /// <summary>
+    /// Gets the metadata key used to store the classification.
+    /// </summary>
+    public static string MetadataKey => "classification";
+
+    /// <inheritdoc />
+    public override async IAsyncEnumerable<IngestionChunk<string>> ProcessAsync(IAsyncEnumerable<IngestionChunk<string>> chunks,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(chunks);
+
+        await foreach (IngestionChunk<string> chunk in chunks.WithCancellation(cancellationToken))
+        {
+            var response = await _chatClient.GetResponseAsync(
+            [
+                new(ChatRole.User,
+                [
+                    _request,
+                    new TextContent(chunk.Content),
+                ])
+            ], _chatOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            chunk.Metadata[MetadataKey] = response.Text;
+
+            yield return chunk;
+        }
+    }
+
+    private static TextContent CreateLlmRequest(ReadOnlySpan<string> predefinedClasses, string fallbackClass)
+    {
+        StringBuilder sb = new("You are a classification expert. Analyze the given text and assign a single, most relevant class. ");
+
+#pragma warning disable IDE0058 // Expression value is never used
+        sb.Append("Use only the following predefined classes: ");
+        for (int i = 0; i < predefinedClasses.Length; i++)
+        {
+            sb.Append(predefinedClasses[i]);
```

Comment:
Are there any constraints on what these can look like? What if they contain input that messes with the prompt?

Reply:
I see, I assumed you expected some kind of prompt injection prevention.

> the same separator you're using, e.g. what if the class they provide is "foo, bar"?

How should I handle such inputs? Throw? Or escape them when building prompt?